### ISSUE TITLE
Build failure: onnxruntime

### ISSUE THREAD

**Issue by elizagamedev** [NixOS/nixpkgs#282465]
### Steps To Reproduce

1. `git checkout 6bd1f8c6ffbb93c3e74ebcf2e80da070d9c1545f`
1. `nix-build -A onnxruntime`

### Build log

[gist link](https://gist.github.com/elizagamedev/1979aab39c765383c5fc54c7c6fa6e95)

### Additional context

N/A

### Notify maintainers

@jonringer @puffnfresh @ck3d @cbourjau 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.72, NixOS, 23.11 (Tapir), 23.11.20240115.b8dd8be`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.05"`
 - channels(eliza): `""`
 - nixpkgs: `/nix/store/yy19v2dwb8ldphvia9smajvwv3ycx2c1-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by pinpox** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/258392

**Comment by yshui** [comment]
In addition, if you build with `cudaSupport = true`, the derivation doesn't even evaluate:

```
       … while evaluating derivation 'onnxruntime-1.16.3'
         whose name attribute is located at /nix/store/cbbsvzjcqvzlylq9s7p6d2gyxh64q88p-source/pkgs/stdenv/generic/make-derivation.nix:353:7

       … while evaluating attribute 'cmakeFlags' of derivation 'onnxruntime-1.16.3'
         at /nix/store/cbbsvzjcqvzlylq9s7p6d2gyxh64q88p-source/pkgs/stdenv/generic/make-derivation.nix:417:7:
          416|
          417|       cmakeFlags =
             |       ^
          418|         cmakeFlags

       … from call site
         at /nix/store/cbbsvzjcqvzlylq9s7p6d2gyxh64q88p-source/pkgs/development/libraries/onnxruntime/default.nix:190:6:
          189|   ] ++ lib.optionals cudaSupport [
          190|     (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_CUTLASS" cutlass)
             |      ^
          191|     (lib.cmakeFeature "onnxruntime_CUDNN_HOME" cudaPackages.cudnn)

       … while calling 'cmakeFeature'
         at /nix/store/cbbsvzjcqvzlylq9s7p6d2gyxh64q88p-source/lib/strings.nix:812:27:
          811|   */
          812|   cmakeFeature = feature: value:
             |                           ^
          813|     assert (lib.isString feature);

       error: assertion '((lib).isString value)' failed
       at /nix/store/cbbsvzjcqvzlylq9s7p6d2gyxh64q88p-source/lib/strings.nix:814:5:
          813|     assert (lib.isString feature);
          814|     assert (lib.isString value);
             |     ^
          815|     cmakeOptionType "string" feature value;
```

**Comment by autrimpo** [comment]
The evaluation error has been fixed in #291874, are there any other build failures with the current version?

**Comment by pinpox** [comment]
Builds fine on current master, this can be closed.
